### PR TITLE
Fix Ui Canvas Invalid Render Target Crash

### LINKED ISSUE #10429: Ui Canvas Crashes If Render Target is Empty
**Describe the bug**
The Editor will crash in game mode when using a Ui Canvas on Mesh for a ui canvas that hasn't set the render target.

**Assets required**
[!Important!] Please provide sample assets needed to reproduce the issue, either as an [attachment](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files) or as a link to a public asset in Github.

**Steps to reproduce **
1. Create a new ui canvas
2. Enable "Render to Texture" on the canvas, but don't define a Render Target
image

3. Open a new level and create an entity with components:
    a. UI Canvas Asset Ref
    b. Mesh
    c. Ui Canvas on Mesh
image
4. Press CTRL+G

**Expected behavior**
Game runs without crashing

**Actual behavior**
Crash

image

>	LyShine.Tools.dll!UiCanvasComponent::UseRenderTargetAsset(const AZ::Data::Asset<AZ::RPI::AttachmentImageAsset> & attachmentImageAsset) Line 1908	C++
 	[Inline Frame] LyShine.Tools.dll!UiCanvasComponent::CreateRenderTarget() Line 3706	C++
 	LyShine.Tools.dll!UiCanvasComponent::RenderCanvas(bool isInGame, AZ::Vector2 viewportSize, UiRenderer * uiRenderer) Line 2009	C++
 	LyShine.Tools.dll!UiCanvasManager::RenderLoadedCanvases() Line 628	C++
 	LyShine.Tools.dll!CLyShine::Render() Line 440	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::Internal::INVOKE(void(AZ::RPI::ViewportContextNotifications::*)() &) Line 177	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::invoke(void(AZ::RPI::ViewportContextNotifications::*)() &) Line 54	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZ::EBusEventProcessingPolicy::Call(void(AZ::RPI::ViewportContextNotifications::*)() &) Line 437	C++
 	Atom_RPI.Editor.dll!AZ::Internal::EBusContainer<AZ::RPI::ViewportContextNotifications,AZ::RPI::NotifyByViewportNameTraits,1,1>::Dispatcher<AZ::EBus<AZ::RPI::ViewportContextNotifications,AZ::RPI::NotifyByViewportNameTraits>>::Event<void (__cdecl AZ::RPI::ViewportContextNotifications::*)(void)>(const AZ::Name & id, void(AZ::RPI::ViewportContextNotifications::*)() && func) Line 146	C++
 	Atom_RPI.Editor.dll!AZ::RPI::ViewportContext::OnBeginPrepareRender() Line 132	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::Internal::INVOKE(void(AZ::RPI::SceneNotification::*)() &) Line 177	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZStd::invoke(void(AZ::RPI::SceneNotification::*)() &) Line 54	C++
 	[Inline Frame] Atom_RPI.Editor.dll!AZ::EBusEventProcessingPolicy::Call(void(AZ::RPI::SceneNotification::*)() &) Line 437	C++
 	Atom_RPI.Editor.dll!AZ::Internal::EBusContainer<AZ::RPI::SceneNotification,AZ::RPI::SceneNotification,1,1>::Dispatcher<AZ::EBus<AZ::RPI::SceneNotification,AZ::RPI::SceneNotification>>::Event<void (__cdecl AZ::RPI::SceneNotification::*)(void)>(const AZ::Uuid & id, void(AZ::RPI::SceneNotification::*)() && func) Line 146	C++
 	Atom_RPI.Editor.dll!AZ::RPI::Scene::PrepareRender(AZ::RHI::JobPolicy jobPolicy, float simulationTime) Line 685	C++
 	Atom_RPI.Editor.dll!AZ::RPI::RPISystem::RenderTick() Line 315	C++
 	[Inline Frame] EditorLib.dll!AZStd::Internal::INVOKE(void(AZ::SystemTickEvents::*)() &) Line 177	C++
 	[Inline Frame] EditorLib.dll!AZStd::invoke(void(AZ::SystemTickEvents::*)() &) Line 54	C++
 	[Inline Frame] EditorLib.dll!AZ::EBusEventProcessingPolicy::Call(void(AZ::SystemTickEvents::*)() &) Line 437	C++
 	EditorLib.dll!AZ::Internal::EBusContainer<AZ::SystemTickEvents,AZ::SystemTickEvents,0,1>::Dispatcher<AZ::EBus<AZ::SystemTickEvents,AZ::SystemTickEvents>>::Broadcast<void (__cdecl AZ::SystemTickEvents::*)(void)>(void(AZ::SystemTickEvents::*)() && func) Line 1360	C++
 	EditorLib.dll!AZ::ComponentApplication::TickSystem() Line 1357	C++
 	EditorLib.dll!CCryEditApp::IdleProcessing(bool bBackgroundUpdate) Line 2282	C++
 	EditorLib.dll!CCryEditApp::OnIdle(long lCount) Line 2192	C++
 	EditorLib.dll!Editor::EditorQtApplication::maybeProcessIdle() Line 312	C++
 	[External Code]	
 	EditorLib.dll!CryEditMain(int argc, char * * argv) Line 3940	C++
 	[Inline Frame] Editor.exe!AZStd::Internal::INVOKE(int(*)(int, char * *) &) Line 208	C++
 	[Inline Frame] Editor.exe!AZStd::invoke(int(*)(int, char * *) &) Line 54	C++
 	Editor.exe!main(int argc, char * * argv) Line 32	C++
 	Editor.exe!WinMain(HINSTANCE__ * __formal, HINSTANCE__ * __formal, char * __formal, int __formal) Line 97	C++

**Found in Branch**
developement latest

**Desktop/Device (please complete the following information):**
 - OS: Windows

### PR DESCRIPTION
**Pull request by AMZN-Gene**
Stop runtime crash by making sure Ui Canvas render target asset is valid before attempting to access its properties
Also added misc minor code cleanup

Tested by running gamemode with a ui canvas that has render targets enabled but has not set a render target .attimage asset and seeing that the game doesn't crash.
Also testing using with a valid render target and seeing that the rendering is successful.

Resolves #10429

Signed-off-by: Gene Walters <genewalt@amazon.com>

### CODE REVIEW

File: Gems/LyShine/Code/Source/UiCanvasComponent.h [hunk 1]
```diff
@@ -230,12 +230,12 @@ class UiCanvasComponent
     void MarkRenderGraphDirty() override;
     // ~UiCanvasComponentImplementationInterface
 
-    // RenderToTextureRequests
+    // LyShine::RenderToTextureRequestBus::Handler
```
Comment (cgalvan) [inline Gems/LyShine/Code/Source/UiCanvasComponent.h:233]: Nitpick: For styling consistency this should be updated to `// LyShine::RenderToTextureRequestBus overrides ...`

File: Gems/LyShine/Code/Source/UiCanvasComponent.h [hunk 2]
```diff
@@ -230,12 +230,12 @@ class UiCanvasComponent
...
     AZ::RHI::AttachmentId UseRenderTarget(const AZ::Name& renderTargetName, AZ::RHI::Size size) override;
     AZ::RHI::AttachmentId UseRenderTargetAsset(const AZ::Data::Asset<AZ::RPI::AttachmentImageAsset>& attachmentImageAsset) override;
     void ReleaseRenderTarget(const AZ::RHI::AttachmentId& attachmentId) override;
     AZ::Data::Instance<AZ::RPI::AttachmentImage> GetRenderTarget(const AZ::RHI::AttachmentId& attachmentId) override;
-    // ~RenderToTextureRequests
+    // ~LyShine::RenderToTextureRequestBus::Handler
```
Comment (cgalvan) [inline Gems/LyShine/Code/Source/UiCanvasComponent.h:238]: Can remove this comment

**Review by cgalvan — APPROVED** [review]

File: Gems/LyShine/Code/Source/UiCanvasComponent.h (hunk 1 above)
Comment (AMZN-Gene) [reply Gems/LyShine/Code/Source/UiCanvasComponent.h]: fixed!

File: Gems/LyShine/Code/Source/UiCanvasComponent.h (hunk 2 above)
Comment (AMZN-Gene) [reply Gems/LyShine/Code/Source/UiCanvasComponent.h]: thanks :D

**Pull request merged**